benchmark/sirun/shimmer-runtime/index.js

```diff
@@ -0,0 +1,66 @@
+const shimmer = require('../../../packages/datadog-shimmer')
+
+const {
+  ENABLED,
+  WRAP_FUNCTION,
+  FUNCTION_NAME,
+} = process.env
+
+const ITERATIONS = 1e8
+
+let counter = 0
+function declared () {
+  return ++counter // Do very little
+}
+
+const arrow = () => {
+  return ++counter // Do very little
+}
+
+async function asyncDeclared () {
+  return ++counter // Do very little
+}
+
+const asyncArrow = async () => {
+  return ++counter // Do very little
+}
+
+let testedFn = {
+  declared,
+  arrow,
+  asyncDeclared,
+  asyncArrow
+}[FUNCTION_NAME]
+if (!testedFn) {
+  throw new Error(`Function ${FUNCTION_NAME} not found`)
+}
+
+if (ENABLED !== 'true') {
+  for (let i = 0; i < ITERATIONS; i++) {
+    testedFn()
+  }
+} else {
+  if (WRAP_FUNCTION === 'true') {
+    const wrapped = shimmer.wrapFunction(testedFn, (original) => {
+      return function () {
+        return original.apply(this, arguments)
```

Comment:
Please add some dc.channels and some minimal if argument checks to pretend there is some work? That way we can measure some actual overhead. Right now I expect the in-between method to pretty much be zero overhead.

Reply:
Ok, makes sense. I would just not fully rely on that. Inlining could for example work magic for this case which it could potentially not do in other cases.